fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -703,19 +696,21 @@ private void handleSetStmt() {
     }
 
     // Process a select statement.
-    private void handleQueryStmt(List<PlanFragment> fragments, List<ScanNode> scanNodes, TDescriptorTable descTable,
-                                 List<String> colNames, List<Expr> outputExprs, String explainString) throws Exception {
+    private void handleQueryStmt(ExecPlan execPlan) throws Exception {
         // Every time set no send flag and clean all data in buffer
         context.getMysqlChannel().reset();
-        StatementBase queryStmt = parsedStmt;
 
-        if (queryStmt.isExplain()) {
-            handleExplainStmt(explainString);
+        if (parsedStmt.isExplain()) {
+            handleExplainStmt(execPlan);
```

Comment:
when context.getSessionVariable().isReportSucc(), also need to handleExplainStmt

Reply:
I'm afraid not. 

Previous code `parsedStmt.isExplain() || context.getSessionVariable().isReportSucc()` means needs to generate `explainString`, but only `parsedStmt.isExplain()` need to response result as `explain sql`.

Reply:
```
        if (context.getQueryDetail() != null) {
            context.getQueryDetail().setExplain(explainString);
        }
```
you should keep explain string in QueryDetail, workGroup also need to be kept in QueryDetail

Reply:
fixed